File: 1_Two Sum.go
//Two Sum value finding index
func twoSum(nums []int, target int) []int {
    mapping := map[int] int{}
    
    for index, value := range nums {
        mapping[value] = index
    }
    
    result := []int {}
    
    for i, value := range nums {
        key := target - value
        
        index, found := mapping[key]
        
        if found && index != i {
            result = append(result, i)
            result = append(result, index)
            break
        }
    }
    
    return result
}

File: 1_Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::vector<int> result;
        std::unordered_map<int, int> map;
        
        for(int i = 0; i < nums.size(); ++i) {
            map[nums[i]] = i;
        }
        
        for(int i = 0; i < nums.size(); ++i) {
            int rest = target - nums[i];
            
            if(map.find(rest) != map.end()) {
                if(i != map[rest]) {
                    result.push_back(i);
                    result.push_back(map[rest]);
                    return result;    
                }
            }
        }
        
        return result;
    }
};